src/jdk.compiler/share/classes/com/sun/tools/javac/parser/JavacParser.java

```diff
@@ -3898,11 +3918,34 @@ public JCTree.JCCompilationUnit parseCompilationUnit() {
                         reportSyntaxError(token.pos, Errors.ExpectedModule);
                     }
                 }
+
                 defs.appendList(semiList.toList());
-                JCTree def = typeDeclaration(mods, docComment);
-                if (def instanceof JCExpressionStatement statement)
-                    def = statement.expr;
-                defs.append(def);
+                boolean isTopLevelMethodOrField = false;
+
+                // Do to a significant number of existing negative tests
+                // this code speculatively tests to see if a top level method
+                // or field can parse. If the method or field can parse then
+                // it is parsed. Otherwise, parsing continues as though
+                // anonymous main classes did not exist and error reporting
+                // is the same as in the past.
+                if (Feature.ANONYMOUS_MAIN_CLASSES.allowedInSource(source) &&
+                        !isDeclaration() &&
+                        (token.kind == VOID || token.kind == IDENTIFIER)) {
```

Comment:
These checks (`token.kind == VOID || token.kind == IDENTIFIER`) for token kind will fail to parse fields of primitive types (or methods with primitive types as a return type)? Either the tests must include the primitive types, or (maybe) just this test can be avoided completely?

Reply:
Changing